### PR TITLE
If 'featured' option is selected, auto-select 'sticky' | #65897

### PR DESCRIPTION
**Pull request by barryhughes**
Per the relevant user stories, if a user checks the 'feature event' checkbox, let's automatically check the 'sticky in month view' checkbox (they can manually de-select if they wish).

Note: rather than cram more code into `events-admin.js` I've created a new file that is only interested in the event post editor with the intent that we add more to it and pull across any other event editor-specific items going forward.

https://central.tri.be/issues/65897

### CODE REVIEW

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -1991,6 +1991,7 @@ public function add_admin_assets() {
 				Tribe__Events__Template_Factory::asset_package( 'ecp-plugins' );
 
 				if ( $admin_helpers->is_post_type_screen( self::POSTTYPE ) ) {
+					Tribe__Events__Template_Factory::asset_package( 'event-editor' );
```
Comment (bordoni) [inline src/Tribe/Main.php]: Can we use the new `tribe_asset` functions? Instead of this one?

File: src/resources/js/event-editor.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+	var $sticky_in_month_view_checkbox = $( 'input[name="EventShowInCalendar"]' ),
+	    $featured_event_checkbox = $( 'input[name="feature_event"]' );
+
+	/**
+	 * If the 'feature event' box is checked, automatically check the
+	 * sticky-in-month-view box also.
+	 */
+	obj.auto_enable_sticky_field = function() {
+		if ( $( this ).prop( 'checked' ) ) {
```
Comment (bordoni) [inline src/resources/js/event-editor.js]: Can we use the inverted logic with `return`?

Comment (barryhughes) [inline src/resources/js/event-editor.js]: Return early if it isn't checked? I mean, we could, but in this case that would be more verbose as we're only interested in performing a single operation when the condition is met.

File: src/Tribe/Main.php (hunk 1 above)
Comment (barryhughes) [inline src/Tribe/Main.php]: Updated!

File: src/resources/js/event-editor.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+var tribe_events_event_editor = tribe_events_event_editor || {};
+
+/**
+ * Implements behaviours that are specific to the event editor.
+ */
+( function( obj, $ ) {
+
+	var $sticky_in_month_view_checkbox = $( 'input[name="EventShowInCalendar"]' ),
```
Comment (borkweb) [inline src/resources/js/event-editor.js]: As long as this file is enqueued in the footer, this should work fine. But in the event that it is moved to the header, the input fields may not exist on the page. My suggestion would be to add an `obj.init()` method and fire that off during document.ready.

**Comment by borkweb** [comment]
⚡️

**Pull request merged**